src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/build_plan/JenkinsPipelineScriptCreator.java

```diff
@@ -120,6 +120,9 @@ private Optional<String> getProjectTypeName(final ProgrammingLanguage programmin
         else if (projectType.isPresent() && projectType.get().isGradle()) {
             return Optional.of("gradle");
         }
+        else if (projectType.isPresent() && projectType.get().equals(ProjectType.MAVEN_BLACKBOX)) {
+            return Optional.of("blackbox");
+        }
```

Comment:
@Strohgelaender This choice selects the relevant subdirectory from which the build script template should be loaded.

Reply:
ok, thanks. So it is an explicit check (I thought this was selecting the template folder when looking through the changes).